docs/developers/smart-contract-tutorial-hardhat.md

```diff
@@ -0,0 +1,177 @@
+
+Next, update the configuration at __hardhat.config.js__ to include the Celestia Ethermint network:
+
+```javascript
+require("@nomiclabs/hardhat-waffle");
+
+module.exports = {
+  solidity: "0.8.13",
+  networks: {
+    hardhat: {},
+    celestiaEthermint: {
+      url: "http://35.208.160.145:8545",
+      accounts: [process.env.PRIVATE_KEY],
+      chainId: 69420
+    }
+  }
+};
+```
+
+### Updating the contracts and tests
+
+Let's update the contracts to include a basic counter example. Create a new file in the __src__ directory named __Counter.sol__ and add the following code:
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+contract Counter {
+    int private count;
+    
+    constructor(int _count) {
+        count = _count;
+    }
+
+    function incrementCounter() public {
+        count += 1;
+    }
+    function decrementCounter() public {
+        count -= 1;
+    }
+
+    function getCount() public view returns (int) {
+        return count;
+    }
+}
+```
+
+Next, let's create a test for this contract.
+
+Open __test/sample-test.js__ and update the code with the following:
+
+```javascript
+const { expect } = require("chai");
+const { ethers } = require("hardhat");
+
+describe("Counter", function () {
+  it("Should increment the counter", async function () {
+    const Counter = await ethers.getContractFactory("Counter");
+    const counter = await Counter.deploy(10);
+    await counter.deployed();
+
+    expect(await counter.getCount()).to.equal(10);
+
+    await counter.incrementCounter();
+
+    expect(await counter.getCount()).to.equal(11);
+
+    await counter.decrementCounter();
+    await counter.decrementCounter();
+
+    expect(await counter.getCount()).to.equal(9);
+  });
+});
+```
+
+Next, we can test the contract with the following command:
+
+```sh
+npx hardhat test
+```
+
+### Updating the deployment script
+
+Now that we've tested the contract, let's try deploying it locally.
+
+To do so, update the deloyment script at __scripts/sample-script.js__ with the following code:
+
+```javascript
+const hre = require("hardhat");
+
+async function main() {
+  const Counter = await hre.ethers.getContractFactory("Counter");
+  const counter = await Counter.deploy(10);
+
+  await counter.deployed();
+  console.log("Counter deployed to:", counter.address);
+}
+
+main()
+  .then(() => process.exit(0))
+  .catch((error) => {
+    console.error(error);
+    process.exit(1);
+  });
+
+```
+
+### Deploying locally
+
+Next start the local testnet:
+
+```sh
+npx hardhat node
+```
+
+Once started, Hardhat will give you a local RPC endpoint as well as a handful of Private Keys and Accounts that you can use.
+
+We can now deploy locally. In a separate terminal window, run the following command:
+
+```sh
+npx hardhat run scripts/sample-script.js --network localhost
+```
+
+You can now interact with the network from your application via the JSON-RPC server at `http://127.0.0.1:8545/`.
```

Comment:
Is there an additional step to testing the contract after this?

Reply:
Not really, unless we want to include how someone might write some JavaScript to interact with the deployment with from their project or a client-side app

One of the things I like the most about Foundry is that `cast` allows you to send calls directly from the CLI, I don't think Hardhat has that capability